cms/navigation/templatetags/navigation_tags.py

```diff
@@ -0,0 +1,104 @@
+from typing import TYPE_CHECKING, Optional, TypedDict
+
+import jinja2
+from django import template
+from django.http import HttpRequest
+
+if TYPE_CHECKING:
+    from wagtail.blocks import StructValue
+
+    from cms.navigation.models import MainMenu
+
+
+register = template.Library()
+
+
+class LinkItem(TypedDict, total=False):
+    text: str
+    url: str
+    children: list["LinkItem"]
+
+
+class ColumnData(TypedDict):
+    column: int
+    linksList: list["LinkItem"]
+
+
+def _extract_highlight_item(value: "StructValue", request: Optional["HttpRequest"] = None) -> dict[str, str]:
```

Comment:
@sanjeevz3009 Might it be simpler/easier to define this sort of thing as a method on the model class?

Reply:
@ababic Made some changes to Jake's comments above and now I have `_extract_item`. Do you still think it may be worth adding `_extract_item` to the model class?

Reply:
Let's keep it in template tags for now. We can revise for footer menu